opentelemetry-sdk/src/resource/builder.rs

```diff
@@ -0,0 +1,108 @@
+use std::time::Duration;
+
+use opentelemetry::{KeyValue, Value};
+
+use super::{Resource, ResourceDetector, SERVICE_NAME};
+
+/// Builder to allow easy composition of a Resource
+#[derive(Debug, Default)]
+pub struct ResourceBuilder {
+    resource: Resource,
+}
+
+impl ResourceBuilder {
+    /// Create ResourceBuilder with an empty [Resource].
+    pub fn new_empty() -> Self {
+        ResourceBuilder {
+            resource: Resource::empty(),
+        }
+    }
+
+    /// Create ResourceBuilder with a default [Resource].
+    pub fn new_default() -> Self {
```

Comment:
```suggestion
    pub fn default() -> Self {
```
Probably should be done as an `impl Default for ResourceBuilder` to make it more generally compatible.

Reply:
updated